cunumeric/module.py

```diff
@@ -2296,22 +2375,25 @@ def indices(dimensions, dtype=int, sparse=False):
     N = len(dimensions)
     shape = (1,) * N
     if sparse:
-        res = tuple()
+        res_tuple: tuple[ndarray, ...] = ()
+        for i, dim in enumerate(dimensions):
+            idx = arange(dim, dtype=dtype).reshape(
+                shape[:i] + (dim,) + shape[i + 1 :]
+            )
+            res_tuple += (idx,)
+        return res_tuple
     else:
         out_shape = (N,) + dimensions
-        res = empty(out_shape, dtype=dtype)
-    for i, dim in enumerate(dimensions):
-        idx = arange(dim, dtype=dtype).reshape(
-            shape[:i] + (dim,) + shape[i + 1 :]
-        )
-        if sparse:
-            res = res + (idx,)
-        else:
-            res[i] = idx
-    return res
+        res_array: ndarray = empty(out_shape, dtype=dtype)
+        for i, dim in enumerate(dimensions):
+            idx = arange(dim, dtype=dtype).reshape(
+                shape[:i] + (dim,) + shape[i + 1 :]
+            )
+            res_array[i] = idx
+        return res_array
```

Comment:
I split things here. In stead of a loop over a conditional, it's now a conditional with separate loops in each branch. This could probably be cleaned up more by moving the branch blocks to helper functions.